src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -168,25 +276,44 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                     // don't resolve builtins like `#[derive]`
                     if let Res::Def(..) = res {
                         let res = res.map_id(|_| panic!("unexpected node_id"));
-                        return Some(res);
+                        return Some(Ok(res));
                     }
                 }
             } else {
                 debug!("attempting to resolve item without parent module: {}", path_str);
+                return Some(Err(ResolutionFailure::NoParentItem));
             }
```

Comment:
At some point I should probably refactor this out so it only happens in one place. The only time this should happen is for synthetic impls or something which can't be documented anyway.

Reply:
I think fixing this should go in a follow-up PR - it's a easy change but it will be a lot of churn, making this harder to review.